src/transaction/id.rs

```diff
@@ -51,7 +51,7 @@ impl Transaction {
                 ..
             } = input
             {
-                utxo_id.iter_mut().for_each(|b| *b = 0);
+                *utxo_id = UtxoId::default();
```

Comment:
This will lead to performance degradation. The previous code will just `memset` while the new code will potentially allocate+copy. Probably a clipper nit?

Reply:
UtxoId now has two field so i used default. But you got me interested in optimization side and for [u8;32] it is not even memset but moveups, and with optimizer turned on it does not matter, it is same for assign, iter and default: https://rust.godbolt.org/z/xhjaqjxWe